File: src/ELibrary.Backend/LibraryShopEntities/Domain/Entities/Shop/CartBook.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LibraryShopEntities.Domain.Entities.Shop
{
    public class CartBook
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public string Id { get; set; } = default!;
        [Required]
        [Range(0, int.MaxValue, ErrorMessage = "Book amount must be greater or equal than 0")]
        public int BookAmount { get; set; }
        [Required]
        public int BookId { get; set; } = default!;
        [Required]
        public string CartId { get; set; } = default!;
        [ForeignKey("CartId")]
        public Cart Cart { get; set; } = default!;

        public void Copy(CartBook other)
        {
            this.BookAmount = other.BookAmount;
        }
    }
}

File: src/ELibrary.Frontend/src/app/modules/shared/domain/models/shop/cart/cartBook.ts
import { Book, getDefaultBook } from "../../../..";

export interface CartBook {
    id: string;
    bookAmount: number;
    bookId: number;
    book: Book;
}

export function getDefaultCartBook(): CartBook {
    return {
        id: "",
        bookAmount: 0,
        bookId: 0,
        book: getDefaultBook()
    }
}